### ISSUE TITLE
Add unit testing support

### ISSUE THREAD

**Issue by feschmidt** [steelelabgit/stlab#17]
We should add some test files to support unit testing, see [here](https://code.visualstudio.com/docs/python/unit-testing)

**Comment by mgely** [comment]
Ideally we want to set it up such that each time we push, github automatically runs all our unit tests and sends an email if something is broken. We don't need to reinvent the wheel here this is very standard practice.
It is called CI (for Continuous Integration) and there are a million companies offering CI services. For open source projects, https://travis-ci.org offers a free service, and seems to be one of the best.
Basically, they copy the latest pushed version to their servers, run the tests, and communicate the result back to github.

**Comment by mgely** [comment]
We don't need to go crazy, but we could implement some basic tests to make sure no one brakes the core code

**Comment by feschmidt** [comment]
CI will probably be very useful once we separate measurement from analysis software, as I don't see an easy way for unit testing of actual hardware
